### PR TITLE
feat: add types for input components

### PR DESCRIPTION
**Pull request by jorenbroekema**
## What I did

1. added types for input components: 
- date
- email
- iban 
- range
- stepper

### CODE REVIEW

File: packages/input-date/src/LionInputDate.js [hunk 1]
```diff
@@ -24,15 +28,19 @@ export class LionInputDate extends LocalizeMixin(LionInput) {
 
   constructor() {
     super();
-    this.parser = (value, options) => (value === '' ? undefined : parseDate(value, options));
+    /**
+     * @param {string} value
+     */
+    this.parser = value => (value === '' ? undefined : parseDate(value));
```
Comment (daKmoR) [inline packages/input-date/src/LionInputDate.js:34]: we no longer parse the options?

File: packages/input-range/src/LionInputRange.js [hunk 2]
```diff
@@ -3,18 +3,31 @@ import { css, html, unsafeCSS } from '@lion/core';
...
+// @ts-expect-error https://github.com/microsoft/TypeScript/issues/40110 + false positive for incompatible static get properties. Lit-element merges super properties already for you.
 export class LionInputRange extends LocalizeMixin(LionInput) {
   static get properties() {
     return {
-      min: Number,
-      max: Number,
-      unit: String,
+      min: {
+        type: Number,
+        reflect: true,
```
Comment (daKmoR) [inline packages/input-range/src/LionInputRange.js:21]: why reflect?

File: packages/input-stepper/src/LionInputStepper.js [hunk 3]
```diff
@@ -22,30 +22,53 @@ export class LionInputStepper extends LionInput {
...
     return {
-      min: Number,
-      max: Number,
-      step: Number,
-      modelValue: Number,
-      __disableIncrementor: Boolean,
-      __disableDecrementor: Boolean,
+      min: {
+        type: Number,
+        reflect: true,
```
Comment (daKmoR) [inline packages/input-stepper/src/LionInputStepper.js:27]: why reflect?

File: packages/input-range/src/LionInputRange.js (hunk 2 above)
Comment (jorenbroekema) [reply packages/input-range/src/LionInputRange.js:21]: If a user changes the `min` attribute, it should reflect to the property and vice versa.

File: packages/input-date/src/LionInputDate.js (hunk 1 above)
Comment (jorenbroekema) [reply packages/input-date/src/LionInputDate.js:34]: parseDate does not accept any 🙈  it uses localize.locale. There's no other parse options like there is with numbers.

Comment (daKmoR) [reply packages/input-date/src/LionInputDate.js:34]: hmm ok, but `parseDate` should accept options like a different locale... 😅

but yeah that is for another pull request

**Review by daKmoR — APPROVED** [review]

**Pull request merged**